libcst/codemod/commands/convert_type_comments.py

```diff
@@ -623,10 +642,31 @@ def leave_With(
     # (B) we also manually reach down to the first statement inside of the
     #     funciton body and aggressively strip type comments from leading
     #     whitespaces
+    #
+    # PEP 484 underspecifies how to apply type comments to (non-static)
+    # methods - it would be possible to provide a type for `self`, or to omit
+    # it. We use heuristics to determine when a function is a method, and allow
+    # for the `self` or `cls` argument to have no type provided.
+    #
+    # The heuristics are:
+    # - Is it inside of a class body? If not, it is never a method.
+    # - Is the name of the first parameter `cls` or `self`? This isn't required
+    #   but it's a strong enough convention that it should work as a heuristic.
 
-    def visit_FunctionDef(
+    def visit_Class(
         self,
         node: cst.FunctionDef,
```

Comment:
How is this not a type error?
```suggestion
        node: cst.Class,
```

Reply:
Woah.

I think it might be because the visitor pattern, as far as I can tell, is entirely dynamic so pyre may have no idea what's actually going on.